src/exercises/exerciseTypes/clickWordInContext/ClickWordInContext.js

```diff
@@ -0,0 +1,185 @@
+import { useState, useEffect, useContext } from "react";
+import * as s from "../Exercise.sc.js";
+
+import BottomInput from "../findWordInContext/BottomInput.js";
+
+import strings from "../../../i18n/definitions";
+import NextNavigation from "../NextNavigation";
+import LoadingAnimation from "../../../components/LoadingAnimation.js";
+import InteractiveText from "../../../reader/InteractiveText.js";
+import { TranslatableText } from "../../../reader/TranslatableText.js";
+import { tokenize } from "../../../utils/preprocessing/preprocessing";
+import { SpeechContext } from "../../../contexts/SpeechContext.js";
+import useSubSessionTimer from "../../../hooks/useSubSessionTimer.js";
+
+const EXERCISE_TYPE = "Click_L1W_in_L2T";
+export default function ClickWordInContext({
+  api,
+  bookmarksToStudy,
+  correctAnswer,
+  notifyIncorrectAnswer,
+  setExerciseType,
+  isCorrect,
+  setIsCorrect,
+  moveToNextExercise,
+  toggleShow,
+  reload,
+  setReload,
+  exerciseSessionId,
+  activeSessionDuration,
+}) {
+  const [messageToAPI, setMessageToAPI] = useState("");
+  const [articleInfo, setArticleInfo] = useState();
+  const [interactiveText, setInteractiveText] = useState();
+  const [translatedWords, setTranslatedWords] = useState([]);
+  const speech = useContext(SpeechContext);
+  const [getCurrentSubSessionDuration] = useSubSessionTimer(
+    activeSessionDuration,
+  );
+
+  useEffect(() => {
+    setExerciseType(EXERCISE_TYPE);
+    api.getArticleInfo(bookmarksToStudy[0].article_id, (articleInfo) => {
+      setInteractiveText(
+        new InteractiveText(
+          bookmarksToStudy[0].context,
+          articleInfo,
+          api,
+          "TRANSLATE WORDS IN EXERCISE",
+          EXERCISE_TYPE,
+          speech,
+        ),
+      );
+      setArticleInfo(articleInfo);
+    });
+  }, []);
+
+  useEffect(() => {
+    checkTranslations(translatedWords);
+  }, [translatedWords]);
+
+  function equalAfterRemovingSpecialCharacters(a, b) {
```

Comment:
This function is used in both ClickWordInContext and FindWordInContext we might want to define it outside to avoid duplication.

Reply:
good point!

Reply:
Good point! Actually, FindWordInContext and ClickWordInContext are identical except for the BottomInput. I've created a higher-order component WordInContextExercise and made BottomInput a boolean prop.